### DISCUSSION TITLE
Is there some architecure/structure overview for the scripts?

### DISCUSSION THREAD

**Discussion by soupdiver** [tteck/Proxmox#1340] [General]
Hey,
I just stumbled upon your script collection and on the first glance it looks interesting and useful :)
I was clicking through the scripts a bit and wondered if there is some high-level overview about the structure of the scripts, which files are loaded, bootstrapping, configuration etc.
I don't just want to run some scripts in my setup but want to get an idea of how those work before.

**Comment by tteck** [comment]
In the case of the Docker LXC, the process involves running multiple scripts for each application or service. Firstly, the `docker.sh` script is executed to gather the system parameters. Next, the `build.func` script adds user settings and integrates all the collected information. Afterward, the `create_lxc.sh` script builds the LXC container. Following that, the `docker-install.sh` script is executed, which utilizes the functions exported from the `install.func` script for installing the required applications. Finally, the process returns to the `docker.sh` script to display the completion message.
 **It is essential to thoroughly evaluate the `docker-install.sh` script for a better understanding of the app installation process.**

**Reply by tteck** [reply]
Every application installation uses the same set of scripts, `build.func`, `create_lxc.sh`, and `install.func`. These scripts are reusable and are not specific to any particular application.

**Reply by soupdiver** [reply]
`It is essential to thoroughly evaluate the docker-install.sh script for a better understanding of the app installation process.`

yea, I think that's what I was looking for :D Some not-reading-the-code explanation of what the scripts do. 
But I guess the code is the documentation
